unit/java_bytecode/generate_concrete_generic_type/generate_java_generic_type.cpp

```diff
@@ -160,3 +160,37 @@ SCENARIO(
             "java::java.lang.Integer");
   }
 }
+
+SCENARIO(
+  "generate_java_generic_type_with_array_concrete_type",
+  "[core][java_bytecode][generate_java_generic_type]")
+{
+  auto test_class=
+    "java::generic_field_array_instantiation$generic<java::array[reference]>";
+
+  GIVEN("A generic type instantiated with an array type")
+  {
+    symbol_tablet new_symbol_table=
+      load_java_class(
+        "generic_field_array_instantiation",
+        "./java_bytecode/generate_concrete_generic_type");
+
+    REQUIRE(new_symbol_table.has_symbol(test_class));
+    const symbolt test_class_symbol=
+      new_symbol_table.lookup_ref(test_class);
+
+    REQUIRE(test_class_symbol.type.id()==ID_struct);
+    const typet test_field_type=
+      to_struct_type(test_class_symbol.type).components()[3].type();
```

Comment:
Nit: use `require_type::require_component` to get the specific components type (not obvious that  `components()[3]` corresponds to `f`

Reply:
In fact, is it actually pointing to `af` if the type is pointer to `array[reference]`?